Add documentation for NASweb integration (#28516)
home-assistant/home-assistant.github.io: source/_integrations/nasweb.markdown

---
title: NASweb
description: Integrate NASweb devices
ha_category:
  - Switch
ha_release: '2024.11'
ha_codeowners:
  - '@nasWebio'
ha_iot_class: Local Push
ha_domain: nasweb
featured: true
ha_config_flow: true
ha_platforms:
  - switch
ha_integration_type: hub
---

The **NASweb** {% term integration %} brings device's smart home features into **Home Assistant**, converting them into {% term entities %} that users can monitor, control, and incorporate into {% term scripts %} and {% term automations %}.

NASweb combines the functions of a control panel and the ability to manage building automation. The device monitors the flow of information from sensors and programmable switches and stores settings, definitions and configured actions.\

> [MORE INFO](https://www.chomtech.pl/produkt/naswebio-multisystemowy-sterownik-automatyki-budynkowej/)

Reviewer: ```suggestion
The **NASweb** {% term integration %} brings the device's smart home features into **Home Assistant**, converting them into {% term entities %} that users can monitor, control, and incorporate into {% term scripts %} and {% term automations %}.

NASweb combines the functions of a control panel and the ability to manage building automation. The device monitors the flow of information from sensors and programmable switches and stores settings, definitions, and configured actions. [More information.](https://www.chomtech.pl/produkt/naswebio-multisystemowy-sterownik-automatyki-budynkowej/)
```

## Prerequisites

Before you can add the NASweb integration to Home Assistant, you need to create a user that has access to the NASweb API.

To create a NASweb user with access to the NASweb API, follow thses steps:

1. To log into device `NASweb` page, enter the device IP in your local network:
   - `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)

> 2. In the top-right corner, select your user and in the pop-up, select **Administrators**

Reviewer 2: ```suggestion
2. In the top-right corner, select your user and in the pop-up, select **Administrators**.
```

> 2. In the top-right corner, select your user and in the pop-up, select **Administrators** :

Reviewer 2: ```suggestion
2. In the top-right corner, select your user and in the pop-up, select **Administrators**
```

> 1. To log into device `NASweb` page, enter the device IP in your local network:

Reviewer: That isn't code, so it shouldn't be code fenced.

```suggestion
1. To log into device **NASweb** page, enter the device IP in your local network:
```

>    - `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)

Reviewer: ```suggestion
   - `https://<Device IP address>/nasweb` (for example: `https://192.168.117.230/nasweb`)
```

> 2. In the top-right corner, select your user and in the pop-up, select **Administrators**.
> 
>     ![](/images/integrations/nasweb/dashboard.png)
> 3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.
>     - The `admin` user cannot be used for this. Select another user.
> 
>     ![](/images/integrations/nasweb/users.png)

Reviewer 2: ```suggestion
2. In the top-right corner, select your user and in the pop-up, select **Administrators**.
    ![](/images/integrations/nasweb/dashboard.png)
3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.
    - The `admin` user cannot be used for this. Select another user.
    ![](/images/integrations/nasweb/users.png)
```

> 4. In the **Modules** column, find the **API** module.

Reviewer 2: ```suggestion
4. In the left **Modules** column, find the **API** module.
```

> 5. Select the **API** module and move it into the right-hand column (`1`), which lists modules enabled for this user. 

Reviewer 2: ```suggestion
5. Select the **API** module and drag it into the right-hand column (`1`), which lists modules enabled for this user. 
```

    - To save changes, select **Apply** (`2`).
    ![Administrator User Details](/images/integrations/nasweb/modules.png)

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:
  description: NASweb device address

User:

>   description: Login of user with enabled `API` module

Reviewer: ```suggestion
  description: Login of user with enabled API module
```

Password:

>   description: Password of user with enabled `API` module

Reviewer: ```suggestion
  description: Password of user with enabled API module
```

---
title: NASweb
description: Integrate NASweb devices
ha_category:
  - Switch
ha_release: '2024.11'
ha_codeowners:
  - '@nasWebio'
ha_iot_class: Local Push
ha_domain: nasweb
featured: true
ha_config_flow: true
ha_platforms:
  - switch
ha_integration_type: hub
---

>   ![](/images/integrations/nasweb/nasweb_scheme.png)

Reviewer 2: ```suggestion
```

NASweb combines the functions of a control panel and the ability to manage building automation. The device monitors the flow of information from sensors and programmable switches and stores settings, definitions and configured actions.\
[MORE INFO](https://www.chomtech.pl/produkt/naswebio-multisystemowy-sterownik-automatyki-budynkowej/)

> 

Reviewer 2: ```suggestion

  ![](/images/integrations/nasweb/nasweb_scheme.png)

```

## Prerequisites

Before you can add the NASweb integration to Home Assistant, you need to create a user that has access to the NASweb API.

To create a NASweb user with access to the NASweb API, follow thses steps:

1. To log into device `NASweb` page, enter the device IP in your local network:
   - `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)
2. In the top-right corner, select your user and in the pop-up, select **Administrators** :

    ![](/images/integrations/nasweb/dashboard.png)
3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.
    - The `admin` user cannot be used for this. Select another user.

    ![](/images/integrations/nasweb/users.png)
4. In the **Modules** column, find the **API** module.
5. Select the **API** module and move it into the right-hand column (`1`), which lists modules enabled for this user. 
    - To save changes, select **Apply** (`2`).

> 
>   ![](/images/integrations/nasweb/modules.png)

Reviewer 2: ```suggestion
    ![](/images/integrations/nasweb/modules.png)
```

---
title: NASweb
description: Integrate NASweb devices
ha_category:
  - Switch
ha_release: '2024.11'
ha_codeowners:
  - '@nasWebio'
ha_iot_class: Local Push
ha_domain: nasweb
featured: true
ha_config_flow: true
ha_platforms:
  - switch
ha_integration_type: hub
---

  ![](/images/integrations/nasweb/nasweb_scheme.png)

NASweb combines the functions of a control panel and the ability to manage building automation. The device monitors the flow of information from sensors and programmable switches and stores settings, definitions and configured actions.\
[MORE INFO](https://www.chomtech.pl/produkt/naswebio-multisystemowy-sterownik-automatyki-budynkowej/)

## Prerequisites

Before you can add the NASweb integration to Home Assistant, you need to create a user that has access to the NASweb API.

To create a NASweb user with access to the NASweb API, follow thses steps:

1. To log into device `NASweb` page, enter the device IP in your local network:
   - `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)
2. In the top-right corner, select your user and in the pop-up, select **Administrators** :

>   ![](/images/integrations/nasweb/dashboard.png)
> 
> 3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.

Reviewer 2: ```suggestion
    ![](/images/integrations/nasweb/dashboard.png)
3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.
```

    - The `admin` user cannot be used for this. Select another user.

>   ![](/images/integrations/nasweb/users.png)
> 

Reviewer 2: ```suggestion
    ![](/images/integrations/nasweb/users.png)
```

---
title: NASweb
description: Integrate NASweb devices
ha_category:
  - Switch
ha_release: '2024.11'
ha_codeowners:
  - '@nasWebio'
ha_iot_class: Local Push
ha_domain: nasweb
featured: true
ha_config_flow: true
ha_platforms:
  - switch
ha_integration_type: hub
---

  ![](/images/integrations/nasweb/nasweb_scheme.png)

NASweb combines the functions of a control panel and the ability to manage building automation. The device monitors the flow of information from sensors and programmable switches and stores settings, definitions and configured actions.\
[MORE INFO](https://www.chomtech.pl/produkt/naswebio-multisystemowy-sterownik-automatyki-budynkowej/)

> ## NASweb device configuration for Home Assistant

Reviewer 2: ```suggestion
## Prerequisites
```

> 

Reviewer 2: ```suggestion

Before you can add the NASweb integration to Home Assistant, you need to create a user that has access to the NASweb API.
```

> Log into device `NASweb` page:

Reviewer 2: ```suggestion

To create a NASweb user with access to the NASweb API, follow thses steps:

1. To log into device `NASweb` page, enter the device IP in your local network:
   - `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)
```

> 
> `https://` + `Device IP` + `/nasweb` (for example: `https://192.168.117.230/nasweb`)
> 

Reviewer 2: ```suggestion
```

> Click on your user in top-right corner and select `Administrators` on pop-up user panel:

Reviewer 2: ```suggestion
2. In the top-right corner, select your user and in the pop-up, select **Administrators** :
```

  ![](/images/integrations/nasweb/dashboard.png)

> Find user which will be used by Home Assistant to access NASweb API and click `EDIT` button:

Reviewer 2: ```suggestion
3. Find the user you want to have access to the NASweb API from Home Assistant. Select **Edit**.
    - The `admin` user cannot be used for this. Select another user.
```

> 
> <div class='note warning'>
> 
> User `admin` cannot be used for this purpose, select user with different username
> 
> </div>

Reviewer 2: ```suggestion
```

  ![](/images/integrations/nasweb/users.png)

> In `Modules` left-hand column lists available modules for this user, find `API` module and click it to move it into right-hand column (`1`), which lists modules enabled for this user, then press `APPLY` (`2`) to save changes:

Reviewer 2: ```suggestion
4. In the **Modules** column, find the **API** module.
5. Select the **API** module and move it into the right-hand column (`1`), which lists modules enabled for this user. 
    - To save changes, select **Apply** (`2`).
```
